From 477985a327ff36f91f3509f9f1d175b88c8a4682 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:32 -0400
Subject: [tools] perf/completion: Introduce a layer of indirection

Message-id: <1408953210-26343-25-git-send-email-jolsa@redhat.com>
Patchwork-id: 88766
O-Subject: [PATCH RHEL7.1 BZ1133083 024/322] perf completion: Introduce a layer of indirection
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 2cf025e69543f5f4aa68e8549d60680515fef5ad
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Sun Nov 17 21:43:23 2013 +0530

description
===========
Define the variables cur, words, cword, and prev outside the main
completion function so that we have a chance to override it when we
introduce zsh support.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/bash_completion b/tools/perf/bash_completion
index 62e157d..3efdc84 100644
--- a/tools/perf/bash_completion
+++ b/tools/perf/bash_completion
@@ -89,15 +89,12 @@ __ltrim_colon_completions()
  fi
 }
 
-type perf &>/dev/null &&
-_perf()
+__perf_main ()
 {
- local cur words cword prev cmd
-
- COMPREPLY=()
- _get_comp_words_by_ref -n =: cur words cword prev
+ local cmd
 
  cmd=${words[0]}
+ COMPREPLY=()
 
  # List perf subcommands or long options
  if [ $cword -eq 1 ]; then
@@ -120,6 +117,14 @@ _perf()
   opts=$($cmd $subcmd --list-opts)
   COMPREPLY=( $( compgen -W '$opts' -- "$cur" ) )
  fi
+}
+
+type perf &>/dev/null &&
+_perf()
+{
+ local cur words cword prev
+ _get_comp_words_by_ref -n =: cur words cword prev
+ __perf_main
 } &&
 
 complete -o bashdefault -o default -o nospace -F _perf perf 2>/dev/null \
-- 
1.7.1